### PR TITLE
build: macOS toolchain simplification and bump

### PR DESCRIPTION
**Pull request by dongcarl**
This PR achieves 3 main things:
1. It simplifies the macOS SDK generation by putting the logic inside a (semi-)portable python3 script `gen-sdk`
2. It transitions us to using `libc++` headers extracted from the `Xcode.app`, which is more correct as those headers better match the `.tbd` library stubs we use from the `MacOSX.sdk` (located under the same `Xcode.app`). Previously, we used `libc++` headers copied from our downloaded, pinned clang (see `native_cctools.mk`).
3. It bumps the macOS toolchain in a way that fulfills all of the following constraints:
    1. The new SDK should support compiling with C++17 (our current one doesn't)
    2. The new toolchain should not change our minimum supported macOS version (`-mmacosx-version-min`)
    3. The new toolchain should expect to use a version of `cctools` that is supported by https://github.com/tpoechtrager/cctools-port

For the constraints in (3), you can reference [this chart](https://en.wikipedia.org/wiki/Xcode#Xcode_7.0_-_11.x_(since_Free_On-Device_Development)) to see that the newest toolchain we can use with our `cctools-port` is `11.3.1`, and the rest of the constraints were tested with local builds.

#### But [the other Wikipedia chart](https://en.wikipedia.org/wiki/Xcode#Xcode_11.x_(since_SwiftUI_framework)) says that the "min macOS to run" for Xcode 11.3.1 is 10.14.4, doesn't that violate constraint (ii)?

This confused me at first too, but the "min macOS to run" is for the Xcode.app App itself. The SDK still supports 10.12, as evident in a few plist files and as proven through local builds.

#### Why bundle all of this together in a single PR?

We need (1) and (2) together, because if we don't, manually adding the `libc++` headers and writing that out in a `README.md` is going to result in a lot of user error, so it's great to have these together to be more correct and also make it easier on the user at the same time.

We need (3) together with everything else because bumping (or in the case of (1), renaming) the SDK requires some human coordination and may break some builds. And since it's not that complicated a change, it makes sense to do it together with the rest.

### CODE REVIEW

**Comment by laanwj** [comment]
> For anyone who's looked at the last PR, this one just has a simpler sdk-gen.sh script that I wrote myself, and includes some .xip unpacking innovations thanks to theuni.

Thanks for working on this. This is great news. Maybe can manage to do it myself this time without a Mac :+1:

File: contrib/macdeploy/README.md
Comment (laanwj) [inline contrib/macdeploy/README.md:54]: I'm not sure keeping instructions around for previous SDKs is that useful. Only for the release that is used for building. Older information can always be found in git.

Comment (dongcarl) [reply contrib/macdeploy/README.md]: Fixed!

File: contrib/macdeploy/README.md
Comment (theuni) [inline contrib/macdeploy/README.md:46]: "with the XCODEDIR environment variable pointing to...",
or
"./contrib/macdeploy/gen_sdk_package.sh '/path/to/Xcode.app'"

Which is it? :)

File: contrib/macdeploy/gen-sdk.sh
Comment (theuni) [inline contrib/macdeploy/gen-sdk.sh:8]: Can we name this something to indicate that we've included the headers as well?

**Review by theuni — COMMENTED** [review]
Thanks for doing this. Testing now.

File: contrib/macdeploy/README.md
Comment (theuni) [inline contrib/macdeploy/README.md:46]: gen-sdk.sh != gen-sdk.sh
(below as well)

**Comment by theuni** [comment]
I'm seeing a Qt build failure due to some symlink/hardlink issue:
```
kernel/qdnslookup_unix.cpp:52:10: fatal error: 'arpa/nameser.h' file not found
#include <arpa/nameser.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
Makefile:12337: recipe for target '.obj/qdnslookup_unix.o' failed
```
The link is broken hardlink in the generated sdk:
```
 $ ls -al MacOSX10.14.sdk/usr/include/arpa/nameser.h
lrwxrwxrwx 1 cory cory 26 Jun 12 14:28 MacOSX10.14.sdk/usr/include/arpa/nameser.h -> MacOSX10.14.sdk./nameser.h
```
Whereas in the original SDK it's a working symlink:
```
 $ ls -al MacOSX10.14.sdk/usr/include/arpa/nameser.h
lrwxrwxrwx 1 cory cory 12 Jun 12 14:28 MacOSX10.14.sdk/usr/include/arpa/nameser.h -> ../nameser.h
```

**Review by fanquake — COMMENTED** [review]
Concept ACK.

Cool. Looks like Travis is happy now. Can you squash the fixup commits? 

I ran some gitian builds locally to compare the binaries produced by master (195822f1e05e2f36002c906667d4c639663f23b5) and master + this diff:
```diff
diff --git a/contrib/gitian-descriptors/gitian-osx.yml b/contrib/gitian-descriptors/gitian-osx.yml
index bbae7201e..936d097d2 100644
--- a/contrib/gitian-descriptors/gitian-osx.yml
+++ b/contrib/gitian-descriptors/gitian-osx.yml
@@ -32,7 +32,7 @@ remotes:
 - "url": "https://github.com/bitcoin/bitcoin.git"
   "dir": "bitcoin"
 files:
-- "MacOSX10.14.sdk.tar.gz"
+- "MacOSX10.14-with-libcxx-headers.sdk.tar.gz"
 script: |
   set -e -o pipefail
 
@@ -90,7 +90,7 @@ script: |
   BASEPREFIX="${PWD}/depends"
 
   mkdir -p ${BASEPREFIX}/SDKs
-  tar -C ${BASEPREFIX}/SDKs -xf ${BUILD_DIR}/MacOSX10.14.sdk.tar.gz
+  tar -C ${BASEPREFIX}/SDKs -xf ${BUILD_DIR}/MacOSX10.14-with-libcxx-headers.sdk.tar.gz
 
   # Build dependencies for each host
   for i in $HOSTS; do
diff --git a/depends/hosts/darwin.mk b/depends/hosts/darwin.mk
index 82e086a32..70baeae90 100644
--- a/depends/hosts/darwin.mk
+++ b/depends/hosts/darwin.mk
@@ -1,6 +1,6 @@
 OSX_MIN_VERSION=10.12
 OSX_SDK_VERSION=10.14
-OSX_SDK=$(SDK_PATH)/MacOSX$(OSX_SDK_VERSION).sdk
+OSX_SDK=$(SDK_PATH)/MacOSX$(OSX_SDK_VERSION)-with-libcxx-headers.sdk
 darwin_CC=clang -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK)
 darwin_CXX=clang++ -target $(host) -mmacosx-version-min=$(OSX_MIN_VERSION) --sysroot $(OSX_SDK) -stdlib=libc++
 
diff --git a/depends/packages/native_cctools.mk b/depends/packages/native_cctools.mk
index 4195230b4..19fcf54f9 100644
--- a/depends/packages/native_cctools.mk
+++ b/depends/packages/native_cctools.mk
@@ -73,6 +73,5 @@ define $(package)_stage_cmds
   cp lib/libLTO.so $($(package)_staging_prefix_dir)/lib/ && \
   cp -rf lib/clang/$($(package)_clang_version)/include/* $($(package)_staging_prefix_dir)/lib/clang/$($(package)_clang_version)/include/ && \
   cp bin/llvm-dsymutil $($(package)_staging_prefix_dir)/bin/$(host)-dsymutil && \
-  if `test -d include/c++/`; then cp -rf include/c++/ $($(package)_staging_prefix_dir)/include/; fi && \
   if `test -d lib/c++/`; then cp -rf lib/c++/ $($(package)_staging_prefix_dir)/lib/; fi
```

The binaries produced by the "new" SDK are all slightly larger (except for `bitcoin-wallet`) than the binaries currently produced by master:

| Commit | `bitcoind` | `bitcoin-cli` | `bitcoin-qt` | `bitcoin-tx` | `bitcoin-wallet` | `test_bitcoin` |
| -------- | -------- | ---------- | --------- | ---------- | -------------- | ------------ | 
| Master | 9084960 | 399464 | 24920412 | 1042272 | 2721068 | 15010424 |
| This PR | 9135728 | 407680 | 24958868 | 1050504 | 2717020 | 15021368 |

I did a quick comparison of `bitcoind`, there is a single symbols difference:
```bash
--- 19.txt
+++ ea.txt
@@ -26,14 +26,15 @@
 U std::bad_exception::what() const
 U std::runtime_error::what() const
 U std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::find(char, unsigned long) const
 U std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::rfind(char, unsigned long) const
 U std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::compare(char const*) const
 U std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::compare(unsigned long, unsigned long, char const*) const
 U std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::compare(unsigned long, unsigned long, char const*, unsigned long) const
+T std::__1::enable_if<__can_be_converted_to_string_view<char, std::__1::char_traits<char>, std::__1::basic_string_view<char, std::__1::char_traits<char> > >::value, int>::type std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::compare<std::__1::basic_string_view<char, std::__1::char_traits<char> > >(std::__1::basic_string_view<char, std::__1::char_traits<char> > const&) const
 U std::__1::__shared_weak_count::__get_deleter(std::type_info const&) const
 U std::__1::__vector_base_common<true>::__throw_length_error() const
 U std::__1::__vector_base_common<true>::__throw_out_of_range() const
```

There is also one less initializer function called in the new `bitcoind`, however I haven't looked into that yet.

**Comment by dongcarl** [comment]
Squashed in the fixups!

**Review by Sjors — COMMENTED** [review]
Concept ACK. I did some testing with 1944c5ff798d149422bf6ac7deb999789eef64d2 on macOS and Ubuntu. Code looks reasonable, but mostly above my head.

I found `extract_xcode.py` to be deterministic, but `gen-sdk.sh` produces a different checksum depending on the OS.

File: contrib/macdeploy/README.md
Comment (dongcarl) [reply contrib/macdeploy/README.md:35]: Nice!

File: contrib/macdeploy/README.md
Comment (dongcarl) [reply contrib/macdeploy/README.md:52]: That's somewhat expected, as bsdtar and gnutar (and even different versions of gnutar) might lay out the archive in different ways that all conform to standard. :relaxed:

Comment (Sjors) [reply contrib/macdeploy/README.md:52]: If _you_ can live with a reproducibility issue, it must be fine :-)

**Comment by theuni** [comment]
I discussed this today with @fanquake, and we realized that after this change, we'll also need to bump our SDK in the near-future so that we can begin testing with c++17. And when we do that, it's also a good time to bump our clang/cctools to match.

I hacked that together and tested quickly, and all seems to be working as expected:
https://github.com/theuni/bitcoin/tree/19240

At the risk of slowing down this PR, we could potentially integrate the rest of the toolchain bump in here as well. The upside of that would be that builders only have to download and generate a new sdk tarball once, whereas if we do this in stages, they'll have do it twice.

But then again, Carl has been maintaining these changes for quite a while now and is probably tired of waiting.

@dongcarl Your call :)

File: depends/packages/native_cctools.mk
Comment (theuni) [inline depends/packages/native_cctools.mk:76]: This can be deleted now too, we know where to expect the c++ libs now.

**Comment by Sjors** [comment]
Let's merge this first. As long as the other stuff gets in before 0.21 most people won't be impacted anyway.

**Comment by dongcarl** [comment]
Thought a bit about what's needed for the steps that are required before 0.21, and it seems like I'll have to redo the entire `gen-sdk.sh` in python for the `tarfile` and `plistlib` modules, so it probably doesn't make sense to merge this in the meantime. Will push up a new version when I have it.

Many thanks to laanwj, theuni, fanquake, and especially Sjors for reviewing.

**Comment by theuni** [comment]
> Let's merge this first.

Sounds good.

My only remaining ask for this PR is something @dongcarl and I discussed yesterday: Adding the xcode version string to the (already lengthy!) sdk tarball/dirname. This is to compensate for the fact that Apple updates the SDK in new XCode versions, so unfortunately "MacOSX10.xy.sdk" alone doesn't mean much. Adding this string will prevent future clashes on our end.

**Comment by theuni** [comment]
@dongcarl Whoa, that was quick!

I haven't really reviewed the new script, but it worked great. Built locally, looks good. I went ahead and uploaded the resulting tarball for Travis so we can see where we stand.

Now that this includes a toolchain bump, mind updating the PR title/description? I think this is ready for in-depth review now.

File: contrib/macdeploy/gen-sdk
Comment (Sjors) [inline contrib/macdeploy/gen-sdk:31]: This is not Python 3.5.6 syntax (try with `pyenv`)

**Review by Sjors — APPROVED** [review]
tACK 538bb0b (some minor issues for followup or if you need to touch things)

Gitian build:
```
019e21e451bdba8b803431ddf3be2bc8b2213da02d0163a23b2eda77a6ecb25e  bitcoin-538bb0bbad8d-osx-unsigned.dmg
fc5ea07e47b37adc24513225edc93ddc6a8f6867ae0f24049a560e5a2dd0883d  bitcoin-538bb0bbad8d-osx-unsigned.tar.gz
714bc66d4d7f0131dedd50b5cdfb6da4f1161da91bc2bbeda19f81787ff5756d  bitcoin-538bb0bbad8d-osx64.tar.gz
4958ef9e024dcf41044c0297b278e04b54e422710c6ad813c9bb5b162bd25148  src/bitcoin-538bb0bbad8d.tar.gz
4f5d9ba4ad9e0828cdb7e0c85587d51faf7850be5b8fc68d1b00fe3ccdf7efb3  bitcoin-core-osx-0.21-res.yml
```

Lightly tested DMG on macOS 10.15.5

File: contrib/macdeploy/README.md
Comment (fanquake) [reply contrib/macdeploy/README.md:22]: >  Also, maybe use 11.5?

NACK. 11.3.1 was picked specifically. It's what is referenced throughout the docs, build  system and  in the tarball name. There's also no guarantee that you'll generate the same tarball between various versions of Xcode.

Comment (Sjors) [reply contrib/macdeploy/README.md:22]: What do you mean by "picked specifically"? It's introduced in this PR. The only other place I see it is on the Travis macOS native machine, but the choice there doesn't matter.

Comment (theuni) [reply contrib/macdeploy/README.md:22]: Sorry for not explaining in more detail in the commit message.

11.3.1 was chosen as the last Xcode version which uses (Apple's version of) clang 8.x.x, for which we have a matching cctools and ld64. Starting with 11.4, they bumped to clang 9.x.x. Once cctools-port has a working matching version for clang 9, we can bump to a newer Xcode version.

Comment (dongcarl) [reply contrib/macdeploy/README.md:22]: A thought: to prevent future confusion, we should follow up this PR with documentation on the reasoning behind how our macOS toolchain is constructed.

Comment (Sjors) [reply contrib/macdeploy/README.md:22]: More info would be a good followup. I'm surprised Apple would bump clang 8 to 9 in a minor Xcode version update. But actually that could make sense, because they want to give developers access to SDK's for their next generation(s) OS, without forcing devs to upgrade their production machine to a beta/recent macOS. According to [wikipedia](https://en.wikipedia.org/wiki/Xcode#11.x_series) Xcode 11.3.1 requires macOs 10.14.4, whereas Xcode 11.4 bumps that requirement to macOS 10.15.2. The specific choice of clang 8.x.x makes sense.

File: contrib/macdeploy/gen-sdk
Comment (dongcarl) [reply contrib/macdeploy/gen-sdk:31]: Shame :frowning_face:, I like f-strings! Will redo with `.format`.

Comment (dongcarl) [reply contrib/macdeploy/gen-sdk]: Done and tested with Python 3.5.9!

File: contrib/macdeploy/gen-sdk
Comment (Sjors) [inline contrib/macdeploy/gen-sdk:48]: This should use `{xcode_version}`, not `{version}`; it broke when you changed the formatting.

**Review by Sjors — CHANGES_REQUESTED** [review]
4a7932285ca7fc9a8e91af58567106b2ff04519d is almost there...

File: contrib/macdeploy/gen-sdk
Comment (dongcarl) [reply contrib/macdeploy/gen-sdk]: Good catch! Fixed.

**Comment by Sjors** [comment]
re-tACK cbdaa3a179c5011b651cb4c9cbcef482816d822d

Gitian build changed, presumable because you rebased (from a79bca2f1fb25f433d6e100a31a3acfde2656ce1 to febe5823b4ae0de5cd0e6da3f69555acd0724267)?

```
2f13a3069c1608f0e3c5ab0fa561e22273165acf710c8cc2fe399790b4fd56de  bitcoin-cbdaa3a179c5-osx-unsigned.dmg
fb00a5c1306c33fc54e3aafde51cdc12ab890785eda253717b3924529a6d2366  bitcoin-cbdaa3a179c5-osx-unsigned.tar.gz
4ea413de539ee7e972a5f4d60108016729d7cb92e6d15d5a2747b083fae87140  bitcoin-cbdaa3a179c5-osx64.tar.gz
1173147fd75c59bd44724b489280ea7895b61946a3c91f410d98b5770aa6250a  src/bitcoin-cbdaa3a179c5.tar.gz
dae1d7de69df0f8deb72a0138c74fcb32a4d70a4358e99b78ae2fa362834e4bb  bitcoin-core-osx-0.21-res.yml
```

**Comment by dongcarl** [comment]
I can reproduce Sjor's results!

```
- out_manifest: |
    2f13a3069c1608f0e3c5ab0fa561e22273165acf710c8cc2fe399790b4fd56de  bitcoin-cbdaa3a179c5-osx-unsigned.dmg
    fb00a5c1306c33fc54e3aafde51cdc12ab890785eda253717b3924529a6d2366  bitcoin-cbdaa3a179c5-osx-unsigned.tar.gz
    4ea413de539ee7e972a5f4d60108016729d7cb92e6d15d5a2747b083fae87140  bitcoin-cbdaa3a179c5-osx64.tar.gz
    1173147fd75c59bd44724b489280ea7895b61946a3c91f410d98b5770aa6250a  src/bitcoin-cbdaa3a179c5.tar.gz
```

**Comment by MarcoFalke** [comment]
Let me know when this is ready so I can put the backdoored binary on DrahtBot :)

**Comment by MarcoFalke** [comment]
Before merge, the OP should be edited to remove the historic description, which is accessible through the GitHub history drop-down

**Comment by MarcoFalke** [comment]
the ci failure can be ignored: https://github.com/bitcoin/bitcoin/issues/19329#issuecomment-646979700

**Comment by MarcoFalke** [comment]
Concept ACK cbdaa3a179c5011b651cb4c9cbcef482816d822d (I read the OP and don't disagree)

**Comment by dongcarl** [comment]
@MarcoFalke theuni said he'll run thru it once more on Monday, should be pretty safe after that!

File: depends/packages/native_cctools.mk
Comment (fanquake) [inline depends/packages/native_cctools.mk:2]:
```suggestion
$(package)_version=4da2f3b485bcf4cef526f30c0b8c0bcda99cdbb4
```

**Review by fanquake — CHANGES_REQUESTED** [review]
Looks good. Seems to align with what me and @theuni discussed. Tested `extract_xcode.py`. However, I noticed while building depends that the version of LD64 being passed around didn't look correct `-DLD64_VERSION_NUM=512.4`. It should be `530`. Looks like upstream didn't update it when doing the last LD64 update, however they have noticed and fixed it, so we should move to cctools @ https://github.com/tpoechtrager/cctools-port/commit/4da2f3b485bcf4cef526f30c0b8c0bcda99cdbb4, which is in the `949.0.1-ld64-530` branch.

File: depends/packages/native_cctools.mk
Comment (dongcarl) [reply depends/packages/native_cctools.mk]: Done!

**Comment by dongcarl** [comment]
2418f739f75824d6689369f326b960cec254cf56 includes fanquake's recommendation [here](https://github.com/bitcoin/bitcoin/pull/19240#discussion_r443187584), and a modification that makes the history more readable (some leftover mistake from my squashing).

Should be ready to merge once theuni takes a last look today.

File: depends/packages/native_cctools.mk
Comment (theuni) [reply depends/packages/native_cctools.mk]: Nice spot, thanks.

**Comment by theuni** [comment]
Actually, after digging through @fanquake's recommendation some more, it looks like the LD64_VERSION_NUM does **not** end up being passed through correctly. We can poke at clang's source to see what it does with that value. One easy-to-demonstrate case can be seen [here](https://github.com/llvm/llvm-project/blob/master/clang/lib/Driver/ToolChains/Darwin.cpp#L217)
```c++
  // Newer linkers support -demangle. Pass it if supported and not disabled by
  // the user.
  if (Version[0] >= 100 && !Args.hasArg(options::OPT_Z_Xlinker__no_demangle))
    CmdArgs.push_back("-demangle");
```

Linking a test with our depends-generated toolchain:
> /home/cory/dev/bitcoin2/depends/x86_64-apple-darwin16/native/bin/clang++ -target x86_64-apple-darwin16 -mmacosx-version-min=10.12 --sysroot /home/cory/dev/bitcoin2/depends/SDKs/Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers -stdlib=libc++ -std=c++11 test.cpp -o testing -v

Shows the link-line:
> "/home/cory/dev/bitcoin2/depends/x86_64-apple-darwin16/native/bin/x86_64-apple-darwin16-ld" -dynamic -arch x86_64 -macosx_version_min 10.12.0 -syslibroot /home/cory/dev/bitcoin2/depends/SDKs/Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers -o testing /tmp/test-385d92.o -lc++ -lSystem

However, a test with -mlinker-version=100:

> /home/cory/dev/bitcoin2/depends/x86_64-apple-darwin16/native/bin/clang++ -target x86_64-apple-darwin16 -mmacosx-version-min=10.12 --sysroot /home/cory/dev/bitcoin2/depends/SDKs/Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers -stdlib=libc++ -std=c++11 test.cpp -o testing -v -mlinker-version=100


Yields:
> "/home/cory/dev/bitcoin2/depends/x86_64-apple-darwin16/native/bin/x86_64-apple-darwin16-ld" **-demangle** -dynamic -arch x86_64 -macosx_version_min 10.12.0 -syslibroot /home/cory/dev/bitcoin2/depends/SDKs/Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers -o testing /tmp/test-a0732e.o -lc++ -lSystem

Note that "-demangle" was added as-expected only when -mlinker-version was passed. Without that option, it seems to assume the lowest possible value. So I guess this is a regression from all the way back in ca5055a5aa07aba81a87cf12f6f0526a63c423b5 :(

I suggest we take https://github.com/theuni/bitcoin/commit/718180005fb031713445498d89cc0565e5a6f6ec as part of this PR as well. Then maybe we can send @fanquake down the rabbit hole to see why clang can't find this on its own (or if it's even supposed to).

**Comment by dongcarl** [comment]
cherry-picked 718180005fb031713445498d89cc0565e5a6f6ec into this PR so that we only nuke depends cache once. @theuni anything else you want to address?

**Review by theuni — APPROVED** [review]
ACK adf543d7144a44f8cd28a090a21c4e2a606862da.

Thanks, everyone for sticking with this. The scope ended up expanding a few times, but I'm really happy to get this all in at once.

Big thanks to @Sjors as well for testing along the way.

**Comment by dongcarl** [comment]
Gitian build succeeded locally:
```yml
- out_manifest: |
    5eb9795549bc1f3a91cfa4b8901db357343d6386a68c15298009546ed28f3f74  bitcoin-adf543d7144a-osx-unsigned.dmg
    8f03081ef08035ee91c5fa10e1ddb86842cb318dd62ea923e7346c6f4306cd96  bitcoin-adf543d7144a-osx-unsigned.tar.gz
    f065ab8d37a473c180bb92f7613de03ba3a1b5f604f0aaa2cd4dd8d9953a9e7f  bitcoin-adf543d7144a-osx64.tar.gz
    ffc2e29782439497a2a27de16996d6b3de5225feb06e6f63ece3680557470e37  src/bitcoin-adf543d7144a.tar.gz
```

**Review by fanquake — APPROVED** [review]
ACK adf543d7144a44f8cd28a090a21c4e2a606862da - I'll take a look at the linker issue.

**Comment by Sjors** [comment]
Post merge: I get the same gitian result as @dongcarl found.

**Comment by MarcoFalke** [comment]
Is the `Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers.tar.gz` deterministic and where can I get it without an apple account?

**Comment by MarcoFalke** [comment]
@jonasschnelli https://bitcoin.jonasschnelli.ch/gitian/#nighly also needs the new tar.gz

**Comment by theuni** [comment]
> Is the `Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers.tar.gz` deterministic and where can I get it without an apple account?

The .tar.gz is not deterministic for the usual tarball reasons, but its contents are.

I'm afraid you'll need an apple account to grab Xcode.

**Comment by Sjors** [comment]
@MarcoFalke you can compare your results with mine: https://github.com/bitcoin/bitcoin/pull/19240#discussion_r442705416

**Comment by MarcoFalke** [comment]
Could someone please upload a mac build.log for me to compare? I get a mismatch

**Pull request merged**
